bpf/process/bpf_execve_event.c

```diff
@@ -175,6 +175,16 @@ event_execve(struct sched_execve_args *ctx)
 	event = map_lookup_elem(&execve_msg_heap_map, &zero);
 	if (!event)
 		return 0;
+
+	/*
+	 * At this point on errors proceed with care:
+	 *  1. Ensure that event fields are zeroed or initialized
+	 *  2. On errors update event->process.flags to reflect those
+	 *     errors and proceed collecting _all_ execve event data.
+	 *
+	 * This prevents tail calling execve_send() on an invalid event.
```

Comment:
How about we make this more specific?

"
The event we allocate above includes uninitiated data, which are zeroed out in __event_get_cgroup_info called by __event_get_task_info. If code changes so that we return before __event_get_task_info is called, ensure that the necessary fields are zeroed out and that event->process.flags is set properly.
"

Reply:
Done, thanks!